planex/extract.py

```diff
@@ -0,0 +1,184 @@
+"""
+planex-extract: Extract files from a tarball as described by a link file
+"""
+
+import argparse
+import json
+import logging
+import os
+import os.path
+import re
+import sys
+import tarfile
+import urlparse
+
+import argcomplete
+
+from planex.util import add_common_parser_options
+from planex.util import setup_logging
+from planex.util import setup_sigint_handler
+import planex.spec
+
+
+def all_sources(spec, topdir, check_package_names):
```

Comment:
Please move this into the Spec class.   You can then factor it out of fetch.py as well.